src/qml/BarcodeReader.qml

```diff
@@ -0,0 +1,267 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+import QtQuick.Shapes 1.14
+import QtMultimedia 5.14
+
+import org.qfield 1.0
+
+import Theme 1.0
+
+Popup {
+  id : barcodeReader
+
+  signal decoded(var string)
+
+  property int popupWidth: mainWindow.width <= mainWindow.height ? mainWindow.width - 40 : mainWindow.height - 40
```

Comment:
While not sure if it is worth it for this PR, I believe we should make all this magic numbers Theme variables.   In this case is `popupScreenEdgeMargin` I guess.

Reply:
@suricactus , very good idea sir.